examples/bsp_ext.py

```diff
@@ -0,0 +1,68 @@
+import os
+import os.path
+import ruamel.yaml
+
+
+def action_extensions(base_actions, project_path=os.getcwd()):
+    """ Describes extension for Board Support Packages. """
+
+    def bsp_remove_namespace(bsp):
+        return bsp[len('espressif/'):] if bsp.startswith('espressif/') else bsp
+
+    def bsp_check_callback(action: str, ctx, args, **kwargs: str) -> None:
+        yaml = ruamel.yaml.YAML()
+        bsps = {
+            'esp_wrover_kit'       : 'esp32',
+            'esp32_azure_iot_kit'  : 'esp32',
+            'esp32_s2_kaluga_kit'  : 'esp32s2',
+            'esp32_s3_eye'         : 'esp32s3',
+            'esp32_s3_lcd_ev_board': 'esp32s3',
+            'esp32_s3_usb_otg'     : 'esp32s3',
+            'esp-box'              : 'esp32s3',
+            }
+        IDF_MAIN_MANIFEST = os.path.join(project_path, 'main/idf_component.yml')
+        with open(IDF_MAIN_MANIFEST, 'r+') as f:
+            manifest = yaml.load(f)
+            active_bsps = list()
+            for dep in manifest['dependencies']:
+                dep_without_namespace = bsp_remove_namespace(dep)
+                if dep_without_namespace in bsps:
+                    active_bsps.append(dep)
```

Comment:
I guess the difference is that the ruamel.yaml solution preserves the comments in the yaml file, while the pyyaml one generates the file from scratch. Not sure how important this is for the use case, though.

Reply:
Thank you for the reviews @igrr @kumekay 

To be perfectly honest, I used the first module that I got after googling 'edit yaml file python' :D

Anyway, IMHO, this module is better for the job (compared to `pyyaml`) because it allows me to _edit_ rather than _recreate_ the file. I found this useful when handling nested yaml entries (i.e. versions in dependencies).

The extension now creates its own command `set-bsp` rather than hijacking the existing `add-dependency` command, PTAL!